src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/ModelExplorerExtensions.cs

```diff
@@ -40,6 +41,25 @@ public static string GetSimpleDisplayText(this ModelExplorer modelExplorer)
                 return modelExplorer.Metadata.NullDisplayText;
             }
 
+            if (modelExplorer.Metadata.IsEnum)
+            {
+                var modelEnum = (Enum) modelExplorer.Model;
```

Comment:
Changed from suggested code which included the cast in the if statement like this:
`if(modelExplorer.Metedata.IsEnum && _model is Enum modelEnum)`
Rease being, if IsEnum is true and the model value is not an enum, it would be better to throw an exception rather than swallow the error. Thoughts?

Reply:
Do this in the suggested way. `Model` may be a `string` regardless of the datatype. And, that's not an error,

Reply:
"not an error" - is that just because of the way the the ModelExplorer works?